nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -208,14 +211,18 @@ def training_step(self, batch, batch_idx):
         else:
             loss_mean = torch.tensor(0.0).cuda()
 
+        # when using sequence parallelism, the sequence parallel layernorm grads must be all-reduced
+        if self.cfg.get('tensor_model_parallel_size', 1) > 1 and self.cfg.get('sequence_parallel', False):
+            self.allreduce_sequence_parallel_gradients()
```

Comment:
> It seems my RETRO model is broken due to [b,s] order swtiched. do we have a sequence gather utility method so I can work on the full sequence of data?

Yes, there's `tensor_parallel.gather_from_sequence_parallel_region()`

Reply:
> what about O1 case? does it create problem if the sequence parallel layernorm grads is all-reduced and use PLT to all reduce all the grads.

I don't think this will be an issue with O1. We still need this reduction.